Neo4j.Driver/Neo4j.Driver/Internal/IO/ChunkReader.cs

```diff
@@ -76,47 +68,93 @@ private bool TryReadFromBuffer(Stream targetStream)
                     }
                 }
 
-                if (_currentChunkSize != -1 && HasBytesAvailable(_currentChunkSize + _chunkSizeBuffer.Length))
+                // As long as we know the chunk size and some bytes available in the buffers, write those
+                // to the target stream.
+                if (_currentChunkSize != -1 && HasBytesAvailable())
                 {
-                    CopyToFromChunkStream(targetStream, _currentChunkSize);
+                    var count = Math.Min(_currentChunkSize, _lastWritePosition - _lastReadPosition);
+
+                    CopyFromBuffer(targetStream, count);
```

Comment:
Renamed ```targetStream``` to ```messageStream``` but did not touch ```downStream``` as it may not be networkStream which we're dealing with.